### PR TITLE
improve validations in rbe_autoconfig

### PR DESCRIPTION
**Pull request by nlopezgi**
* Add validations to check when docker image will be needed to be pulled
* Split off logic to resolve project_root from main impl method
* Produce a dummy file if an image was pulled (for tests)
* Add tests that validate an image will not be pulled in edge case
* Add tests to valiate image was pulled in other cases

### CODE REVIEW

File: rules/rbe_repo.bzl [hunk 1]
```diff
@@ -416,6 +436,10 @@ def _pull_image(ctx, docker_tool_path, image_name):
     _print_exec_results("pull image", result, fail_on_error = True)
     ctx.report_progress("image pulled.")
 
+    # Create a dummy file with the image name to enable testing
+    # if container was pulled
+    ctx.file("image_name", image_name, False)
```
Comment (xingao267) [inline rules/rbe_repo.bzl]: Write some message to the file to indicate this file is just to indicating if the container is pulled or not?

Comment (nlopezgi) [reply rules/rbe_repo.bzl]: done
